com.eclipsesource.modelserver.client/src/main/java/com/eclipsesource/modelserver/client/ModelServerClient.java

```diff
@@ -0,0 +1,167 @@
+package com.eclipsesource.modelserver.client;
+
+import com.eclipsesource.modelserver.common.ModelServerPaths;
+import com.eclipsesource.modelserver.jsonschema.Json;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import okhttp3.*;
+import org.jetbrains.annotations.NotNull;
+
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionException;
+
+public class ModelServerClient implements ModelServerClientApi, ModelServerPaths {
+
+    private OkHttpClient client;
+    private String baseUrl;
+
+    ModelServerClient(String baseUrl) throws MalformedURLException {
+        this(new OkHttpClient(), baseUrl);
+    }
+
+    ModelServerClient(OkHttpClient client, String baseUrl) throws MalformedURLException {
+        this.client = client;
+        this.baseUrl = new URL(baseUrl).toString();
+    }
+
+    @Override
+    public CompletableFuture<Response<String>> get(String modelUri) {
+        final Request request = new Request.Builder()
+            .url(makeUrl(MODEL_CRUD).replace(":modeluri", modelUri))
+            .build();
+
+        return makeCall(request).thenApply(response -> parseField(response, "data"));
+    }
+
+    @Override
+    public CompletableFuture<Response<List<String>>> getAll() {
+        final Request request = new Request.Builder()
+            .url(makeUrl(MODEL_URIS))
+            .build();
+
+        return makeCall(request)
+            .thenApply(response -> parseField(response, "data"))
+            .thenApply(response -> response.mapBody(body -> {
+                List<String> uris = new ArrayList<>();
+                try {
+                    Json.parse(body).forEach(uri -> uris.add(uri.textValue()));
+                    return uris;
+                } catch (IOException e) {
+                    throw new CompletionException(e);
+                }
+            }));
+    }
+
+    @Override
+    public CompletableFuture<Response<Boolean>> delete(String modelUri) {
+        final Request request = new Request.Builder()
+            .url(makeUrl(MODEL_CRUD).replace(":modeluri", modelUri))
+            .delete()
+            .build();
+
+        return makeCall(request)
+            .thenApply(response -> parseField(response, "type"))
+            .thenApply(response -> response.mapBody(body -> body.equals("confirm")));
+    }
+
+    @Override
+    public CompletableFuture<Response<String>> update(String modelUri, String updatedModel, String mediaType) {
+        final Request request = new Request.Builder()
+            .url(makeUrl(MODEL_CRUD).replace(":modeluri", modelUri))
+            .patch(RequestBody.create(updatedModel, MediaType.parse(mediaType)))
+            .build();
+
+        return makeCall(request);
```

Comment:
Parsing of data field is missing here:
`makeCall(request).thenApply(response -> parseField(response, "data"));`